modules/aggregation-layers/src/utils/gpu-grid-aggregation/gpu-grid-aggregator.js

```diff
@@ -366,15 +371,15 @@ export default class GPUGridAggregator {
     // log.assert for required options
     log.assert(
       !changeFlags.dataChanged ||
-        (opts.positions &&
+        (opts.attributes &&
           opts.weights &&
-          (!opts.projectPositions || opts.viewport) &&
+          (!opts.projectPositions || opts.moduleSettings.viewport) &&
           opts.cellSize)
     );
     log.assert(!changeFlags.cellSizeChanged || opts.cellSize);
 
     // viewport is needed only when performing screen space aggregation (projectPoints is true)
-    log.assert(!(changeFlags.viewportChanged && projectPoints) || opts.viewport);
+    log.assert(!(changeFlags.viewportChanged && projectPoints) || opts.moduleSettings.viewport);
```

Comment:
Do not use `assert` unless checking for user error. `GPUGridAggregator` is considered an internal component.

Reply:
`validateProps` props method is validating various parameter combinations, once CPU aggregation is moved out and other refactors done, I can remove it altogether, till then this is handy to catch errors. For now commenting out.